### PR TITLE
Add Fit to Window button

### LINKED ISSUE #128: Introduce Fit-To-Window control to mapping toolbar
As a < Player >, I want < to reset the zoom > so that < I can quickly see the whole map >.

# Acceptance criteria

- [ ] New button present on mapping toolbar
- [ ] Clicking the button zooms out to show full extent of map

# UX
See the "fit to window" icon in the toolbar here:
image

# Notes
It's fine to pull in another open source library, if necessary.  It _probably_ isn't necessary just for this feature.

### PR DESCRIPTION
**Pull request by lm93547**
## 🧰 Ticket
Fixes #128

## 🚀 Overview: 
Add control button to fit to window

## 🔗 Link to preview
https://serge-review-zoom-to-fi-ccljlb.herokuapp.com/

## 🖥️ Screenshot
[If the work is UI related then paste a screenshot of the update here.]
image

### CODE REVIEW

**Review by IanMayo — CHANGES_REQUESTED** [review]
Thanks for the update.  

A couple of fixes please:

1. On Chrome/Macos the control doesn't match the existing ones:
image

2. Again on Chrome/Macos the map zooms out too far.  The map should _Fill_ the viewport, not have large borders:
image

**Review by IanMayo — CHANGES_REQUESTED** [review]
Not great.

1. Setting the zoom level does't scale the plot to the viewport. With a shallow browser window, not all of the map is displayed:
image

2. Please don't leave `console.log()` statements in there.  It's quite right to use them during development, but they should be removed when you're happy with the code.
3. Code outside your own module should only be modified with good reason.  Did you need to modify `zoomSnap` and `zoomDelta` ?

File: packages/client/src/Helpers/zoomHome.js [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+        // getting the bounds
+        const bounds = [
+          [opts.bounds[0][0], opts.bounds[0][1]],
+          [opts.bounds[1][0], opts.bounds[1][1]]
+        ]
+
+        // fitting the bounds
+        map.fitBounds(bounds);
+
+        // using set timeout which fitsBounds on maximise click (needs a delay as the CSS transition is not finished)
```
Comment (IanMayo) [inline packages/client/src/Helpers/zoomHome.js:57]: @lm93547  - could you teach me about the delay this CSS transition?

What is the problem we have to overcome? Can you give me a link to where I can learn more?

**Comment by IanMayo** [comment]
Hover action

On my system the Leaflet hover highlighting is for the background to go grey, but it goes pink for the home button.

Maybe the leaflet controls go pink your operating system.

Is it an option to find (via 'inspect' the CSS class (or color) name) for the existing zoom in/out buttons, and copy that for the home button?

**Comment by IanMayo** [comment]
Closing #140 resolves a number of issues in this version.

**Comment by lm93547** [comment]
Hi Ian. The reason the hover goes pink is the zoomHome.js has a hover event set to pink, this was purely to test if the hover was working.

**Pull request closed without merging**